lib/api/model/events.dart

```diff
@@ -74,6 +80,57 @@ class AlertWordsEvent extends Event {
   Map<String, dynamic> toJson() => _$AlertWordsEventToJson(this);
 }
 
+/// A Zulip event of type `user_settings` with op `update`.
+@JsonSerializable(fieldRename: FieldRename.snake)
+class UserSettingsUpdateEvent extends Event {
+  @override
+  @JsonKey(includeToJson: true)
+  String get type => 'user_settings';
+
+  @JsonKey(includeToJson: true)
+  String get op => 'update';
+
+  /// The name of the setting, or null if we don't recognize it.
+  @JsonKey(unknownEnumValue: JsonKey.nullForUndefinedEnumValue)
+  final UserSettingName? property;
+
+  /// The new value, or null if we don't recognize the setting.
+  ///
+  /// The type is `dynamic`, but in handling real events from the API,
+  /// it's safe to cast [value] to the type appropriate for [property]
+  /// (e.g., to say `value as bool` for a boolean setting).
+  /// That's because [UserSettingsUpdateEvent.fromJson] does that check itself,
+  /// and it would have failed if the check didn't pass.
+  @JsonKey(readValue: _readValue)
+  final dynamic value;
+
+  /// [value], with a check that its type corresponds to [property]
+  /// (e.g., `value as bool`).
+  static Object? _readValue(Map json, String key) {
+    final value = json['value'];
+    switch (UserSettingName.fromRawString(json['property'] as String)) {
+      case UserSettingName.displayEmojiReactionUsers:
+        return value as bool;
+      case UserSettingName.twentyFourHourTime:
+        return value as bool;
```

Comment:
nit:
```suggestion
      case UserSettingName.displayEmojiReactionUsers:
      case UserSettingName.twentyFourHourTime:
        return value as bool;
```

(doesn't make much difference now, but when we have dozens of these and the bulk of them are of three types, this way will I think be significantly easier to read)

Reply:
different nit: also put the settings in the same order as they appear elsewhere

(same order within a type, I guess — so that is a downside of the grouped approach)